x/concentrated-liquidity/incentives.go

```diff
@@ -760,14 +760,18 @@ func (k Keeper) collectIncentives(ctx sdk.Context, sender sdk.AccAddress, positi
 	}
 
 	// Send the collected incentives to the position's owner from the pool's address.
-	if err := k.bankKeeper.SendCoins(ctx, pool.GetIncentivesAddress(), sender, collectedIncentivesForPosition); err != nil {
-		return sdk.Coins{}, sdk.Coins{}, err
+	if !collectedIncentivesForPosition.IsZero() {
+		if err := k.bankKeeper.SendCoins(ctx, pool.GetIncentivesAddress(), sender, collectedIncentivesForPosition); err != nil {
+			return sdk.Coins{}, sdk.Coins{}, err
+		}
 	}
 
 	// Send the forfeited incentives to the community pool from the pool's address.
-	err = k.communityPoolKeeper.FundCommunityPool(ctx, forfeitedIncentivesForPosition, pool.GetIncentivesAddress())
-	if err != nil {
-		return sdk.Coins{}, sdk.Coins{}, err
+	if !forfeitedIncentivesForPosition.IsZero() {
+		err = k.communityPoolKeeper.FundCommunityPool(ctx, forfeitedIncentivesForPosition, pool.GetIncentivesAddress())
+		if err != nil {
+			return sdk.Coins{}, sdk.Coins{}, err
+		}
```

Comment:
Similar to the comment above, there is no need to run this logic and emit events if the value is zero.